### PR TITLE
Further refactor debug_copy

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -2927,8 +2927,10 @@ def run_wasm_opt(*args, **kwargs):
   @staticmethod
   def debug_copy(src, dst):
     if DEBUG:
-      shutil.copyfile(src, os.path.join(CANONICAL_TEMP_DIR, 'emdebug-%d-%s' % (Building.debug_copy_counter, dst)))
+      dst = 'emcc-%d-%s' % (Building.debug_copy_counter, dst)
       Building.debug_copy_counter += 1
+      logger.debug('saving debug copy %s' % dst)
```
Comment (sbc100) [inline tools/shared.py]: How about `saving intermediate file: `?

Comment (kripken) [reply tools/shared.py]: I actually like `debug copy` as it is a copy in debug mode... and it fits the name of the function.

If you prefer `saving intermediate file`, would changing the function name to `save_intermediate_file` make sense to you? Seems about the same to me.

Comment (sbc100) [reply tools/shared.py]: My understanding is that intermediates are what compilers normally call these things.

I do  think "def save_intermediate(..)` is more meaningful. 

its also not clear do my why this should be on the Building object now, it should just be a function in `shared.py`?   But this is all nits.  This change is good either way.

Comment (kripken) [reply tools/shared.py]: Ok, changed to `save_intermediate`.

I think this makes sense on Building - it's saving of intermediate build files.

**Pull request merged**